### ISSUE TITLE
[Feature Request] Support mobile attachments

### ISSUE THREAD

**Issue by MahmadSharaf** [jvsteiner/s3-image-uploader#29]
Hello there,

I wonder if it is possible that the plugin supports uploading attachments from mobile.

**Comment by jvsteiner** [comment]
I've had this request before.  I don't know how to do it properly.  It worked at one time, and then it stopped.  There have been some suggestions on how to debug it, which I tried, but didn't manage to get anything working.

**Comment by MahmadSharaf** [comment]
Okay, let me try

**Comment by MahmadSharaf** [comment]
Okay, what I have deduced so far:

1. It is not a mobile related problem. The plugin doesn't handle "Insert attachment" command in general.
2. This command opens a file picker, saves it in the vault, and creates a Wiki-link for it.

Solutions I tried in so far with no luck:
1. Intercept the callback function of the file picker and pass the file to your PasteFunction
2. Override the native command and a custom functionality, which opens the file picker and then pass the file to the PasteFunction

```js
async onload() {
...
...
...
   // Wait for Obsidian to load everything, so we make sure that the "Insert attachment" is created by Obsidian
    this.app.workspace.onLayoutReady(() => {
        // Get the command
        const attachFileCommandDefinition = this.app.commands?.commands?.["editor:attach-file"];
       // Store the original callback
        const attachFileCallback = attachFileCommandDefinition?.editorCallback;
        // Check if the callback is stored as expected
        if (typeof attachFileCallback === "function") {
            // Override the existing command with the new callback
            this.app.commands.removeCommand(attachFileCommandDefinition .id);
            this.addCommand({
                id: "editor:insert-attachment",
                name: "Insert Attachment",
                editorCallback: newCallback,
            });
    }
}
```

I am sharing the progress, in case you have any hint on how to proceed. Meanwhile, I am looking into it.

**Comment by jvsteiner** [comment]
maybe some inspiration here? https://github.com/trganda/obsidian-attachment-management/blob/main/src/main.ts

**Comment by MahmadSharaf** [comment]
Yes, I installed it, but unfortunately it doesn't support attachments via the command. But I will look into that direction, I might get inspired from other plugins.

**Comment by jvsteiner** [comment]
>It is not a mobile related problem. The plugin doesn't handle "Insert attachment" command in general.
this might be the key insight

**Comment by jvsteiner** [comment]
So, I have spent some more time investigating this popular request.  I would like to support it, however, unfortunately, my existing code path is frankly a pigs breakfast.  While it would be easy to make "something happen" it would be a lot harder to support all of the configuration options that are present in the existing upload mechanism.  This would require me to refactor it substantially.  Perhaps I am just looking for justifications for my laziness, but another issue came to mind.  In the existing cut and paste insert workflow, the user has implicitly decided exactly where in the note that the image/file to be uploaded should go - that's where the cursor is at in the current note. If I understand the workflow for mobile, its a bit different.  This would enable it to be added from the share menu, or from the camera roll, etc.  In this case, I don't know _where_ in _which_ note to actually put the link.

For those of you following this issue - do you have any desired behavior, or insight into to how it actually should work?  Remember, I don't need or use this feature, so I don't have any clue how you might want it to work.  Please do explain.

**Comment by MahmadSharaf** [comment]
Thank you for giving it another shot.

Here is my mobile workflow:
1. My daily note contains all my life management, so any input I take mostly taken from my mobile.
2. Sometimes, I need to add a photo taken by mobile camera.
3. So I open the note, insert the cursor at the location
4. Then, execute "Insert Attachment" command either from command palette or from context menu.

The problem is that the plugin doesn't intercept this action. Also, the mobile app doesn't support Copy/Paste or Drag and Drop actions.

What I tried so far, knowing that I have no JS experience nor Obsidian API:
1. Intercept the built-in command function right after the file picker. So that the plugin handles the placement as usual.
2. Replace the built-in function while maintaining the same command name for convenience. For example, in my personal use case, I use slash commander that already has "Insert Attachment" command, also Commander plugin that could place a button to trigger the command anywhere. Although it is just for convenience but not big deal
3. The last option, that I couldn't do at all, is to implement the full functionality, which is a file picker and handle the placement of the attachment as usual

I hope you would have a much better luck, and be able to implement this feature.

**Comment by Yul-007** [comment]
Hello, 
thx a lot for your work.

Another fan of the mobile workflow :), I landed here because the Google Photo Plugin might break down if Google changes the workflow. 

I do not know if it helps you: The Google Photo Plugin (https://github.com/alangrainger/obsidian-google-photos/tree/main) works on Mobile, it uses the Command Palette with a command to insert a Photo.

Have a good week!

**Comment by jvsteiner** [comment]
fixed with #[0.2.30](https://github.com/jvsteiner/s3-image-uploader/releases/tag/0.2.30)

**Comment by jvsteiner** [comment]
Use the regular paperclip attachment button - hopefully it works for you.  It's is working on my iphone, but I don't have an android phone to check.

**Comment by MahmadSharaf** [comment]
This exactly what I wished for. Thanks a million.

I can confirm it is working in Android as well

**Comment by jvsteiner** [comment]
whew - glad it's good

**Comment by MahmadSharaf** [comment]
I am sorry for being a bummer. But the new version interferes with the paste command. `ctrl+v` or `paste` command from the context menu, doesn't do anything if the content is text, but it works as expected if it is an image.

The workaround is to use `ctrl+shift+v`.

I rolled back to version 0.2.14 and it works as expected.

**Comment by jvsteiner** [comment]
released a fix - try again - I was calling preventDefault a bit too early.

**Comment by jvsteiner** [comment]
ok - finally fixed I think

**Comment by MahmadSharaf** [comment]
The issue fixed and working on Android as expected. 
Thank you for your quick fix
